news/_posts/2019-09-11-prestashop-in-2019-and-beyond-part-2-pain-points.md

```diff
@@ -0,0 +1,471 @@
+
+
+
+#### Poor documentation
+
+Back in 2017 when I joined PrestaShop, I was surprised to learn how little technical documentation there was. It was hard getting around the code with so many things to learn and so little sources of knowledge. At that time, documentation was hosted in a Confluence server and was 100% written by PrestaShop employees. There was little to no information for 1.7. There were some articles for 1.6, but many of them were outdated and didn't apply for 1.7 code anyway. If learning how PrestaShop works is already hard as it is even when you have some teammates around you to help you, how on Earth could we expect the community to adopt and contribute to 1.7 with no documentation?
+
+For that reason, last year we launched the [DevDocs Project][devdocs], which contains the technical documentation of PrestaShop 1.7. It's [hosted on GitHub][devdocs-gh] so that anyone can contribute. At the time of writing, this doc already has about 200 articles written by over 60 contributors. However, there are still many subjects that [remain to be documented][devdocs-issues].
+
+## Final words
+
+"This has been a very difficult article to write". That's my poor attempt of an excuse for having taken 6 months to put it out the door, and I sincerely apologize for that.
+ 
+This piece isn't meant to be a "character destruction" for PrestaShop 1.7—quite the opposite. After all, I work for PrestaShop and I'm writing on the official blog, so I'm a little biased. But perfect, pristine, legacy-less software doesn't exist, and it's no news that PrestaShop is not the exception. I think that in the spirit of an open source project, and specially one that aims to become _more_ open, it's important be honest and to tell things as they are. Be proud of our strengths, yes, but also, and above all, acknowledge the things that need improving. Above all, _knowing_ yourself and your weakness is the first step towards doing something meaningful about it.
+
+And that's precisely what we mean to do.
+
+In the next part, things will get more interesting: we'll discuss our long-term vision for PrestaShop's **Future Architecture**, and how it will address many of the problems that we currently face.
+
+---
+
+### About the series
+
+In case you forgot, here are the topics that will be covered during this series:
+
+1. [The Current Architecture][part-1] (or _"Point A – Where we are"_)
+2. Pain Points (or _"What needs to be improved"_)
+3. The Future Architecture (or _"Point B – Where we are going"_)
+4. Connecting the dots (or _"Some ideas on how we'll get there"_)
+
+[introduction]: /news/prestashop-in-2019-and-beyond-introduction/
+[part-1]: /news/prestashop-in-2019-and-beyond-part-1-current-architecture/
+[addons]: https://addons.prestashop.com
+[legacy-layout]: https://github.com/PrestaShop/PrestaShop/blob/1.7.6.0/controllers/admin/AdminLegacyLayoutController.php
```

Comment:
Maybe we can also quote the dedicated issue https://github.com/PrestaShop/PrestaShop/issues/12925 that shows our exploration of the topic and the discussion on how best to tackle this problem ?

Reply:
Done